packages/client/components/ReflectionGroup/RemoteReflection.tsx

```diff
@@ -148,15 +152,39 @@ interface Props {
 
 const RemoteReflection = (props: Props) => {
   const {meeting, reflection, style} = props
-  const {id: reflectionId, content, isDropping} = reflection
-  const {meetingMembers} = meeting
+  const {id: reflectionId, content, isDropping, reflectionGroupId} = reflection
+  const {meetingMembers, spotlightGroup} = meeting
   const remoteDrag = reflection.remoteDrag as DeepNonNullable<
     RemoteReflection_reflection['remoteDrag']
   >
   const ref = useRef<HTMLDivElement>(null)
   const [editorState] = useEditorState(content)
   const timeoutRef = useRef(0)
   const atmosphere = useAtmosphere()
+  const spotlightSearchResults = useLazyLoadQuery<RemoteReflectionLocalQuery>(
+    graphql`
+      query RemoteReflectionLocalQuery($reflectionGroupId: ID!, $searchQuery: String!) {
+        viewer {
+          similarReflectionGroups(
+            reflectionGroupId: $reflectionGroupId
+            searchQuery: $searchQuery
+          ) {
+            id
+          }
+        }
+      }
+    `,
+    // TODO: add search query
+    {reflectionGroupId: spotlightGroup?.id || '', searchQuery: ''},
+    {fetchPolicy: 'store-only'}
+  )
+  const isInViewerSpotlightResults = useMemo(() => {
+    const {viewer} = spotlightSearchResults
+    const {similarReflectionGroups} = viewer
+    if (!similarReflectionGroups) return false
+    return !!similarReflectionGroups.find((group) => group.id === reflectionGroupId)
+  }, [spotlightSearchResults])
+
```

Comment:
As I understand this, you're querying the spotlight search to find out whether or not the RemoteReflection is in the Viewer's spotlight, because positioning/zIndex work slightly different there?

I think if we need this information on a reflection to be able to display it correctly, then we should add it to reflection directly once they are included in the spotlight window. Having the same search query here again seems error prone.

Reply:
That's right - if Alice and Bob both open the Spotlight, we need to know whether Alice's remote Spotlight reflection exists in Bob's results. If it does, the zIndex needs to be higher than the modal zIndex, otherwise it should be less.

> we should add it to reflection directly once they are included in the spotlight window

I'm not sure I follow - are you suggesting that we adds the `groupIds` to Relay client's `reflection` object when we first make the query in `ResultsRoot`?

Reply:
I don't like having to somehow pass `searchQuery` to `RemoteReflection`. This doesn't seem like the right thing to do.
I would either convert this into a `preloadedQuery` and pass the queryRef from `ResultsRoot` or simply pass `isInSimilarReflectionGroups` property down from `ResultsRoot` through `useRemoteLyDraggedCard` to `RemoteReflection`. Either way I don't think `searchQuery` belongs here, but @mattkrick might have a better idea here.

Reply:
In the [Loom](https://www.loom.com/share/ac14b15310f447b5bd4f2ef076f317f0) ~34 mins, Matt mentioned a couple of possible approaches here but said that he'd probably keep it the way it is. It looks like the comment was typed but not sent at the end of the review. I'll resolve this conversation for now.